Docs for Autodetected releases and release markers (#9026)
circleci/circleci-docs: jekyll/_cci2/release/releases-overview.adoc

|No. Only set if you want to disable the cancel release feature for your component.
|===

== Autodetected releases and release markers

CircleCI automatically generates release markers for your deployment pipelines. Release markers provide a lightweight way to log your releases without requiring a full CircleCI releases setup. Release markers create for you a log of all releases in one place, for a clear overview of what has changed, without the need to search through your CI/CD pipelines. Release markers log all new releases in one place and link back to the CI/CD pipelines that caused them. You can use release markers independently, without installing the release agent.

=== Autodetection of releases

> CircleCI looks for the word `deploy` in your job names to detect when a release may be in progress and creates an autodetected release marker.

Reviewer: ```suggestion
CircleCI looks for the word `deploy` in your job names to detect when a release may have happened and creates an autodetected release marker.
```

clarify the release might have been done for two reasons

1. we decide to create the release when the deploy job is success, it means is in a terminal status (the thing already happened)
2. we don't care much about the status of the release, we just log the event but the release may be in progress, success or failure status.

When CircleCI creates a release marker for you automatically, we use the information available in your pipeline configuration to create the following:

* **Component**: CircelCI created a component for the release and uses your project name to create a component name.

> * **Environment integration**: CircleCI creates an enviornment for you, using your job and workflow names to create a relevant name. The environment has the type `AUTODETECTED`.

Reviewer: should we mention that if we do a best effort to determine the environment name, if not possible the environment name would be `autogenerated`, also they can feel free to rename it and that won't cause any issues with the autodetection process

Reviewer: the team discarded the idea of updating the type for now, we are going to allow only the name from the time being
